x-pack/plugin/profiling/src/test/java/org/elasticsearch/xpack/profiling/HostMetadataTests.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.profiling;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+public class HostMetadataTests extends ESTestCase {
+    public void testCreateFromRegularSource() {
+        final String hostID = "1440256254710195396";
+        final String machine = "x86_64";
+        final String provider = "aws";
+        final String region = "eu-west-1";
+        final String instanceType = "md5x.large";
+
+        // tag::noformat
+        HostMetadata host = HostMetadata.fromSource(
+            Map.of(
+                "host.id", hostID,
+                "profiling.host.machine", machine,
+                "profiling.host.tags", new ArrayList<>(Arrays.asList(
```

Comment:
Why do you wrap this once more in an `ArrayList`?

Reply:
Good point. Mostly a left over when trying to initialize an ArrayList (the List type that the response parser generates). Then forgot to remove it when finding Arrays.asList().